sdk/identity/azure-identity/azure/identity/_internal/__init__.py

```diff
@@ -67,13 +69,18 @@ def validate_tenant_id(tenant_id):
 def resolve_tenant(default_tenant, tenant_id=None, **_):
     # type: (str, Optional[str], **Any) -> str
     """Returns the correct tenant for a token request given a credential's configuration"""
-    if (
-        tenant_id is None
-        or default_tenant == "adfs"
+    if tenant_id is None:
+        return default_tenant
+    if (default_tenant == "adfs"
         or os.environ.get(EnvironmentVariables.AZURE_IDENTITY_DISABLE_MULTITENANTAUTH)
     ):
+        _LOGGER.info("A token was request for a different tenant than was configured on the credential, "
+                     "but the configured value was used since multi tenant authentication has been disabled. "
+                     "Configured TenantId: {}, Requested TenantId {}".format(default_tenant, tenant_id))
```

Comment:
nit: we can probably use "tenant ID" instead of "TenantID" for these logs, unless there's a logging searchability reason to use this naming approach
```suggestion
                     "Configured tenant ID: {}, Requested tenant ID {}".format(default_tenant, tenant_id))
```